koku/reporting_common/__init__.py

```diff
@@ -14,6 +14,862 @@
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "project_number",
+        "database_table": "reporting_gcpproject",
+        "provider_column_name": "Project Number",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "project_name",
+        "database_table": "reporting_gcpproject",
+        "provider_column_name": "Project Name",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "project_labels",
+        "database_table": "reporting_gcpproject",
+        "provider_column_name": "Project Labels",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "line_item_type",
+        "database_table": "reporting_gcpcostentrylineitemdaily",
+        "provider_column_name": "Line Item",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "start_time",
+        "database_table": "reporting_gcpcostentrylineitemdaily",
+        "provider_column_name": "Start Time",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "end_time",
+        "database_table": "reporting_gcpcostentrylineitemdaily",
+        "provider_column_name": "End Time",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "measurement_type",
+        "database_table": "reporting_gcpcostentrylineitemdaily",
+        "provider_column_name": "Measurement1",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "consumption",
+        "database_table": "reporting_gcpcostentrylineitemdaily",
+        "provider_column_name": "Measurement1 Total Consumption",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "unit",
+        "database_table": "reporting_gcpcostentrylineitemdaily",
+        "provider_column_name": "Measurement1 Units",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "cost",
+        "database_table": "reporting_gcpcostentrylineitemdaily",
+        "provider_column_name": "Cost",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "currency",
+        "database_table": "reporting_gcpcostentrylineitemdaily",
+        "provider_column_name": "Currency",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+    {
+        "database_column": "description",
+        "database_table": "reporting_gcpcostentrylineitemdaily",
+        "provider_column_name": "Description",
+        "report_type": "GCP",
+        "provider_type": "GCP",
+    },
+]
+
+
+def generate_column_map():
+    """Generate a mapping of provider data columns to db columns."""
```

Comment:
Question: 
If these are to be static structures, then why go to the trouble of storing them in code as the table row data and then rebuild a structure from them? Why not just build out the final structure directly in code and just use that?
Was there a reason for doing it this way that I do not understand?

Reply:
Good question, the reason is because the original structure is more human readable than the generated structure here. @Red-HAP  does that resolve the question?

Reply:
I suppose that is a matter of semantics. 
Just seems like a lot of work to maintain one structure in a form that is unnecessary then rebuild it for use in the app.
From my point of view, It seems more difficult since you're maintaining a format in the code that gets transformed into its usable state. Yet the original form stays resident in the code.
It almost seems like you've taken the table out of the database and just stuck it into the code to be processed at module load.

Reply:
Here's the generated structure for reference, Kevan and I agreed that it was less human readable to edit this than it would be to edit the current structure. @Red-HAP 
```
{
    "reporting_awscostentrybill": {
        "bill/BillingEntity": "billing_resource",
        "bill/BillType": "bill_type",
        "bill/PayerAccountId": "payer_account_id",
        "bill/BillingPeriodStartDate": "billing_period_start",
        "bill/BillingPeriodEndDate": "billing_period_end",
    },
    "reporting_awscostentrylineitem": {
        "bill/InvoiceId": "invoice_id",
        "lineItem/LineItemType": "line_item_type",
        "lineItem/UsageAccountId": "usage_account_id",
        "lineItem/UsageStartDate": "usage_start",
        "lineItem/UsageEndDate": "usage_end",
        "lineItem/ProductCode": "product_code",
        "lineItem/UsageType": "usage_type",
        "lineItem/Operation": "operation",
        "lineItem/AvailabilityZone": "availability_zone",
        "lineItem/ResourceId": "resource_id",
        "lineItem/UsageAmount": "usage_amount",
        "lineItem/NormalizationFactor": "normalization_factor",
        "lineItem/NormalizedUsageAmount": "normalized_usage_amount",
        "lineItem/CurrencyCode": "currency_code",
        "lineItem/UnblendedRate": "unblended_rate",
        "lineItem/UnblendedCost": "unblended_cost",
        "lineItem/BlendedRate": "blended_rate",
        "lineItem/BlendedCost": "blended_cost",
        "lineItem/TaxType": "tax_type",
        "pricing/publicOnDemandCost": "public_on_demand_cost",
        "pricing/publicOnDemandRate": "public_on_demand_rate",
        "reservation/AmortizedUpfrontFeeForBillingPeriod": "reservation_amortized_upfront_fee",
        "reservation/AmortizedUpfrontCostForUsage": "reservation_amortized_upfront_cost_for_usage",
        "reservation/RecurringFeeForUsage": "reservation_recurring_fee_for_usage",
        "reservation/UnusedQuantity": "reservation_unused_quantity",
        "reservation/UnusedRecurringFee": "reservation_unused_recurring_fee",
    },
    "reporting_awscostentrypricing": {"pricing/term": "term", "pricing/unit": "unit"},
    "reporting_awscostentryproduct": {
        "product/sku": "sku",
        "product/ProductName": "product_name",
        "product/productFamily": "product_family",
        "product/servicecode": "service_code",
        "product/region": "region",
        "product/instanceType": "instance_type",
        "product/memory": "memory",
        "product/memory_unit": "memory_unit",
        "product/vcpu": "vcpu",
    },
    "reporting_awscostentryreservation": {
        "reservation/ReservationARN": "reservation_arn",
        "reservation/NumberOfReservations": "number_of_reservations",
        "reservation/UnitsPerReservation": "units_per_reservation",
        "reservation/StartTime": "start_time",
        "reservation/EndTime": "end_time",
    },
    "reporting_azurecostentryproductservice": {
        "InstanceId": "instance_id",
        "ResourceLocation": "resource_location",
        "ConsumedService": "consumed_service",
        "ResourceType": "resource_type",
        "ResourceGroup": "resource_group",
        "AdditionalInfo": "additional_info",
        "ServiceTier": "service_tier",
        "ServiceName": "service_name",
        "ServiceInfo1": "service_info1",
        "ServiceInfo2": "service_info2",
    },
    "reporting_azuremeter": {
        "MeterId": "meter_id",
        "MeterName": "meter_name",
        "MeterCategory": "meter_category",
        "Currency": "currency",
        "MeterSubcategory": "meter_subcategory",
        "MeterRegion": "meter_region",
        "UsageRate": "resource_rate",
        "UnitOfMeasure": "unit_of_measure",
    },
    "reporting_azurecostentrylineitem_daily": {
        "SubscriptionGuid": "subscription_guid",
        "Tags": "tags",
        "UsageDateTime": "usage_date",
        "UsageQuantity": "usage_quantity",
        "PreTaxCost": "pretax_cost",
        "OfferId": "offer_id",
    },
    "reporting_ocpusagereportperiod": {
        "cluster_id": "cluster_id",
        "report_period_start": "report_period_start",
        "report_period_end": "report_period_end",
    },
    "reporting_ocpusagereport": {"interval_start": "interval_start", "interval_end": "interval_end"},
    "reporting_ocpusagelineitem": {
        "namespace": "namespace",
        "pod": "pod",
        "node": "node",
        "pod_usage_cpu_core_seconds": "pod_usage_cpu_core_seconds",
        "pod_request_cpu_core_seconds": "pod_request_cpu_core_seconds",
        "pod_limit_cpu_core_seconds": "pod_limit_cpu_core_seconds",
        "pod_usage_memory_byte_seconds": "pod_usage_memory_byte_seconds",
        "pod_request_memory_byte_seconds": "pod_request_memory_byte_seconds",
        "pod_limit_memory_byte_seconds": "pod_limit_memory_byte_seconds",
        "node_capacity_cpu_cores": "node_capacity_cpu_cores",
        "node_capacity_cpu_core_seconds": "node_capacity_cpu_core_seconds",
        "node_capacity_memory_bytes": "node_capacity_memory_bytes",
        "node_capacity_memory_byte_seconds": "node_capacity_memory_byte_seconds",
        "resource_id": "resource_id",
        "pod_labels": "pod_labels",
    },
    "reporting_ocpstoragelineitem": {
        "pod": "pod",
        "namespace": "namespace",
        "persistentvolumeclaim": "persistentvolumeclaim",
        "persistentvolume": "persistentvolume",
        "storageclass": "storageclass",
        "persistentvolumeclaim_capacity_bytes": "persistentvolumeclaim_capacity_bytes",
        "persistentvolumeclaim_capacity_byte_seconds": "persistentvolumeclaim_capacity_byte_seconds",
        "volume_request_storage_byte_seconds": "volume_request_storage_byte_seconds",
        "persistentvolumeclaim_usage_byte_seconds": "persistentvolumeclaim_usage_byte_seconds",
        "persistentvolume_labels": "persistentvolume_labels",
        "persistentvolumeclaim_labels": "persistentvolumeclaim_labels",
    },
    "reporting_ocpnodelabellineitem": {"node": "node", "node_labels": "node_labels"},
    "reporting_gcpproject": {
        "Project ID": "project_id",
        "Account ID": "account_id",
        "Project Number": "project_number",
        "Project Name": "project_name",
        "Project Labels": "project_labels",
    },
    "reporting_gcpcostentrylineitemdaily": {
        "Line Item": "line_item_type",
        "Start Time": "start_time",
        "End Time": "end_time",
        "Measurement1": "measurement_type",
        "Measurement1 Total Consumption": "consumption",
        "Measurement1 Units": "unit",
        "Cost": "cost",
        "Currency": "currency",
        "Description": "description",
    },
}
```

Reply:
Personally, I disagree for several reasons:

1. The initial definition has data in it that isn't used
2. The initial definition stays in memory along with the processed definition
3. I find tracing the dict very easy to follow as a python developer especially since I would have a complete visual reference to the structure as it wil be used in the code.
4. There's no confusion around trying to also determine what the code is doing manipulating the base structure into the final structure.

That said, I _really_ don't feel like this is significant enough to hold up the review. If you feel that strongly and the tests are passing, then I'll approve.

Reply:
i'll include this change @Red-HAP

Reply:
As I said, I could live with either.  I originally suggested printing it out for the same reason as @Red-HAP.  If we run this once, why don't we just store it like that to start with.

Reply:
@kholdaway @Red-HAP I've added this change now.